agent/lib/kontena/service_pods/creator.rb

```diff
@@ -149,7 +149,7 @@ def volumes_exist?(service_pod)
         volume_manager = Celluloid::Actor[:volume_manager]
         service_pod.volumes.each do |volume|
           if volume['name']
-            return false unless volume_manager.volume_exist?(volume['name'])
```

Comment:
This will crash the `VolumeManager` actor if the volume driver mismatches, which isn't great. That's a external configuration error/conflict, and not something that restarting the `VolumeManager` will fix. The exception is good, though, because the `ServicePods::Create` should fail.

It seems like there's a [`Celluloid::abort`](http://www.rubydoc.info/gems/celluloid/Celluloid#abort-instance_method) that can be used to raise an error in the calling actor without crashing the manager actor. That would probably need a separate wrapper function in the `VolumeManager` to rescue and `abort` with the error?